src/component/actions/SetContactAttrib/SetContactAttribForm.test.tsx

```diff
@@ -0,0 +1,162 @@
+            } = setup({
+                onBindWidget: jest.fn()
+            });
+            const initial = propertyToSearchResult(action as SetContactProperty);
+
+            expect(onBindWidgetMock).toHaveBeenCalledTimes(2);
+            expect(onBindWidgetMock).toHaveBeenCalledWith(expect.any(ConnectedAttribElement));
+            expect(onBindWidgetMock).toHaveBeenCalledWith(expect.any(ConnectedTextInputElement));
+            expect(wrapper.find(ConnectedAttribElement).props()).toEqual({
+                name: 'Attribute',
+                showLabel: true,
+                endpoint: endpoints.fields,
+                helpText: ATTRIB_HELP_TEXT,
+                initial,
+                add: true,
+                required: true
+            });
+            expect(wrapper.find(ConnectedTextInputElement).props()).toEqual({
+                name: 'Value',
+                showLabel: true,
+                value: action.value,
+                helpText: TEXT_INPUT_HELP_TEXT,
+                autocomplete: true
+            });
+        });
+    });
+
+    describe('instance methods', () => {
+        describe('getInitial', () => {
+            it('should return contact field SearchResult', () => {
+                const { wrapper, props: { action }, instance } = setup(
+                    { action: setContactField },
+                    true
+                );
+                const expectedInitial = fieldToSearchResult(action as SetContactField);
+
+                expect(instance.getInitial()).toEqual(expectedInitial);
+            });
+
+            it('should return contact property SearchResult', () => {
+                const { wrapper, props: { action }, instance } = setup({}, true);
+                const expectedInitial = propertyToSearchResult(action as SetContactProperty);
+
+                expect(instance.getInitial()).toEqual(expectedInitial);
+            });
+
+            it('should return null', () => {
+                const { wrapper, instance } = setup(
+                    {
+                        action: {
+                            type: 'send_msg',
+                            uuid: '7a19060d-6d93-4217-a4e5-9cbd479be051',
+                            text: 'maravilloso'
+                        }
+                    } as any,
+                    true
+                );
+
+                expect(instance.getInitial()).toBeNull();
+            });
+        });
+
+        describe('onValid', () => {
+            it('should call updateAction prop with new SetContactField action', () => {
+                const {
+                    wrapper,
+                    instance,
+                    props: { action, updateAction: updateActionMock }
+                } = setup({ updateAction: jest.fn(), action: setContactField }, true);
+                const attribute = fieldToSearchResult(action as SetContactField);
+                const { value } = action;
+                const widgets = {
```

Comment:
everything is fairly straightforward until we need to test widget validation...not sure there's a great way to do so other than mocking-out widgets' state.